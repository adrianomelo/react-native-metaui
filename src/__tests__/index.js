import 'react-native';
import React from 'react';
import {
  Text,
  View
} from 'react-native';
import createComponent from '../create-component';
import renderer from 'react-test-renderer';

test('renders single <Text />', () => {
  const types = {Text};
  const text = {
    type: 'Text',
    props: {
      children: 'Hello World'
    }
  };
  const component = createComponent(text, {types});
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders <View /> with <Text />', () => {
  const types = {Text, View};
  const text = {
    type: 'View',
    props: {
      children: {
        type: 'Text',
        props: {
          children: 'Hello World'
        }
      }
    }
  };
  const component = createComponent(text, {types});
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders <View /> with list of <Text />', () => {
  const types = {Text, View};
  const text = {
    type: 'View',
    props: {
      children: [
        {
          type: 'Text',
          props: {
            key: 0,
            children: 'Hello World'
          }
        }
      ]
    }
  };
  const component = createComponent(text, {types});
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders <View /> with null children', () => {
  const types = {Text, View};
  const text = {
    type: 'View',
    props: {
      children: null
    }
  };
  const component = createComponent(text, {types});
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

test('returns string when input is string', () => {
  const text = '';
  const component = createComponent(text, {});
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

test('empty list -> empty list', () => {
  const component = createComponent([], {});
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

test('undefined -> null', () => {
  const component = createComponent(undefined, {});
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

test('throws exeception when type is not defined', () => {
  expect(() => createComponent({}, {})).toThrow(Error);
});
