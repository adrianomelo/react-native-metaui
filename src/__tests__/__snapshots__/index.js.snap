// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty list -> empty list 1`] = `null`;

exports[`renders <View /> with <Text /> 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    Hello World
  </Text>
</View>
`;

exports[`renders <View /> with list of <Text /> 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    Hello World
  </Text>
</View>
`;

exports[`renders <View /> with null children 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    Hello World
  </Text>
</View>
`;

exports[`renders single <Text /> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
>
  Hello World
</Text>
`;

exports[`returns string when input is string 1`] = `""`;

exports[`undefined -> null 1`] = `null`;
